### PR TITLE
docs: update FAQ

### PR DESCRIPTION
**Pull request by zmb3**
Re-order some of the questions to move related topics closer together, and add a new FAQ that explains how roles are embedded in certificates.

### CODE REVIEW

**Review by stevenGravy — APPROVED** [review]

File: docs/pages/faq.mdx [hunk 1]
```diff
@@ -15,32 +15,25 @@ Fortune 500 companies. It has been through several security audits from
...
-functionality without a net addition of an agent on your system.
+## How is Teleport's Community Edition different from Enterprise?
 
-## Can I use OpenSSH with a Teleport cluster?
+Teleport provides three editions:
 
-Yes, this question comes up often and is related to the previous one. Take a
-look at [Using OpenSSH Guide](enroll-resources/server-access/openssh/openssh-agentless.mdx).
+- Teleport Enterprise
+- Teleport Enterprise Cloud
```
Comment (ptgott) [inline docs/pages/faq.mdx:28]: @garofalof Should we say:
- We have two editions, Teleport Enterprise and Teleport Community Edition
- We have three editions, Teleport Enterprise (Cloud), Teleport Enterprise (Self-Hosted), and Teleport Community Edition

Comment (zmb3) [reply docs/pages/faq.mdx:28]: I'd vote for the latter, the parenthetical "(Cloud)" qualifier drives me crazy.

File: docs/pages/faq.mdx [hunk 2]
```diff
@@ -15,32 +15,25 @@ Fortune 500 companies. It has been through several security audits from
...
+behind-firewall clusters, see [Configure Trusted Clusters](admin-guides/management/admin/trustedclusters.mdx).
 
-For capabilities such as kernel-level logging and user provisioning, we
-recommend Teleport as a drop in replacement for OpenSSH. Since Teleport replaces
-the OpenSSH agent while preserving OpenSSH's functionality, you get more
-functionality without a net addition of an agent on your system.
+## How is Teleport's Community Edition different from Enterprise?
 
-## Can I use OpenSSH with a Teleport cluster?
+Teleport provides three editions:
```
Comment (ptgott) [inline docs/pages/faq.mdx:25]:
```suggestion
Teleport provides two editions:
```

File: docs/pages/faq.mdx [hunk 3]
```diff
@@ -15,32 +15,25 @@ Fortune 500 companies. It has been through several security audits from
...
 
-Yes, this question comes up often and is related to the previous one. Take a
-look at [Using OpenSSH Guide](enroll-resources/server-access/openssh/openssh-agentless.mdx).
+- Teleport Enterprise
+- Teleport Enterprise Cloud
+- Teleport Community Edition
```
Comment (ptgott) [inline docs/pages/faq.mdx:29]:
```suggestion
- Teleport Enterprise
- Teleport Community Edition
```
I'm mostly sure this is how we talk about our editions now (e.g., as shown in the [pricing page](https://goteleport.com/pricing/)). @garofalof please feel free to correct this once you are available.

**Review by ptgott — APPROVED** [review]

File: docs/pages/faq.mdx (hunk 1 above)
Comment (garofalof) [reply docs/pages/faq.mdx]: I tend to agree. I think the origin here is that Marketing wanted Enterprise to be positioned as a singular product, which led to the parenthetical qualifiers, but I'm also not a fan. Let me check if we can move to just stating the 3 editions without parentheses.

**Pull request merged**